### PR TITLE
Issue 2004 gmail mock api

### LINKED ISSUE #2004: mock gmail api + draft tests
For me, the PR is good to merge once Roma and Ioan had a chance to comment on the code.

--------------------------

In the meantime, you can start preparing / working on infrastructure for mock tests. You'll probably need to write a HTTP server similar to Gmail API. Here is something to get you started:

Gmail / Google:
endpoints: https://github.com/FlowCrypt/flowcrypt-browser/blob/90b152a474ace38dd96763cf30c9447026ecad76/test/source/mock/google/google-endpoints.ts

Notice that browser extension has a mechanism to export the JSON response from each message:

```ts
export class GoogleData {

  /**
   * This is the proper way to add messages to mock api for testing:
   *   1) log into flowcrypt.compatibility [at] gmail.com
   *   2) go to Settings -> Inbox and find your message
   *   3) click "download api export"
   *   4) save the json file to exported-messages folder
   */
  private static exportedMsgsPath = './test/source/mock/google/exported-messages/';
```

here are the messages: https://github.com/FlowCrypt/flowcrypt-browser/tree/bfd1f698a3f907c86c62640b216f99b8ecc37af7/test/source/mock/google/exported-messages

Then the mock uses these exports, to provide email search functionality and similar: https://github.com/FlowCrypt/flowcrypt-browser/blob/90b152a474ace38dd96763cf30c9447026ecad76/test/source/mock/google/google-data.ts

_Originally posted by @tomholub in https://github.com/FlowCrypt/flowcrypt-android/issues/1983#issuecomment-1269763138_

### PR DESCRIPTION
**Pull request by DenBond7**
This PR added tests for saving drafts via Gmail API

close #2004 

----------------------------------

**Tests** _(delete all except exactly one)_:
- Tests added or updated

--------------------------------

### To be filled by reviewers

I have reviewed that this PR... _(tick whichever items you personally focused on during this review)_:
- [x] addresses the issue it closes (if any)
- [x] code is readable and understandable
- [x] is accompanied with tests, or tests are not needed
- [x] is free of vulnerabilities

### CODE REVIEW

File: FlowCrypt/src/androidTest/java/com/flowcrypt/email/ui/ComposeScreenDraftGmailAPIFlowTest.kt [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+    onView(withId(R.id.editTextEmailSubject))
+      .check(matches(isDisplayed()))
+      .perform(
+        scrollTo(),
+        click(),
+        typeText(MESSAGE_SUBJECT)
+      )
+
+    Thread.sleep(DraftViewModel.DELAY_TIMEOUT * 2)
+    assertTrue(isDraftSaved)
```
Comment (ioanmo226) [inline FlowCrypt/src/androidTest/java/com/flowcrypt/email/ui/ComposeScreenDraftGmailAPIFlowTest.kt:200]: If I understand correctly, this only tests `isDraftSaved` flag gets true when user edits subject.
I think we need to add several other tests like mentioned in https://github.com/FlowCrypt/flowcrypt-ios/pull/1851?

Comment (DenBond7) [reply FlowCrypt/src/androidTest/java/com/flowcrypt/email/ui/ComposeScreenDraftGmailAPIFlowTest.kt:200]: It tests the following scenario:

1. Open the compose screen
2. Type some text in the subject field
3. After that the app should create a new draft on a server
4. `assertTrue(isDraftSaved)` means that the requested `Gmail API` was called with a successful result and we saved a new draft.

> I think we need to add several other tests like mentioned in https://github.com/FlowCrypt/flowcrypt-ios/pull/1851?

Sorry, it's a little hard to understand `iOs` code for me. 

I just see the following comment

```
By the way, I think we should have UI test(for last case at above comment) which tests draft send feature

Create draft
Go back and open that draft
Update draft(update subject, message, recipient)
Send message
check if sent message is correct
```

Did you mean this one? 

@ioanmo226 Please add examples of other scenarios if you think they should be tested too.

Comment (ioanmo226) [reply FlowCrypt/src/androidTest/java/com/flowcrypt/email/ui/ComposeScreenDraftGmailAPIFlowTest.kt:200]: Please let me know if you have questions.

Comment (DenBond7) [reply FlowCrypt/src/androidTest/java/com/flowcrypt/email/ui/ComposeScreenDraftGmailAPIFlowTest.kt:200]: Thinking... The above scenario looks too complex. It tests too many things that should be split. Also, the current Android realization and Espresso testing will not cover it without `dirty` code. On Android creating/updating a draft is a background task(out of the screen/Activity).

We can split the above scenario into the following
### Check the correct creating and updating
```
1. Create draft
2. Go back and create another draft
3. Check if 2 drafts are saved correctly
4. Update 1st draft. Go back to draft list screen and check if 1st draft is updated correctly and 2nd draft remains same
```

### Check the correct draft deleting
```
1. We have two drafts
2. Delete 1st draft, reload draft screen and check if 1st draft is deleted correctly. 
```

```
1. We have one drafts
2. Delete a draft and reload draft screen to check if draft folder is empty.
```

### Check the correct draft sending
```
1. Create new draft(only input subject) and go back and open that draft again
2. Update subject, insert message, recipients and click send button
3. Check if draft is sent correctly (Check if draft screen is empty after email send, go to sent folder and check if sent email is there)
```

Comment (ioanmo226) [reply FlowCrypt/src/androidTest/java/com/flowcrypt/email/ui/ComposeScreenDraftGmailAPIFlowTest.kt:200]: I agree

Comment (DenBond7) [reply FlowCrypt/src/androidTest/java/com/flowcrypt/email/ui/ComposeScreenDraftGmailAPIFlowTest.kt:200]: I propose to merge the current PR that added a test for creating a draft and add new tests via a separate PR.
@ioanmo226 What do you think?

Comment (ioanmo226) [reply FlowCrypt/src/androidTest/java/com/flowcrypt/email/ui/ComposeScreenDraftGmailAPIFlowTest.kt:200]: I agree

**Review by ioanmo226 — APPROVED** [review]

**Pull request merged**